main.tf

```diff
@@ -483,6 +484,14 @@ resource "azurerm_kubernetes_cluster" "main" {
       condition     = var.network_plugin_mode != "Overlay" || var.network_plugin == "azure"
       error_message = "When network_plugin_mode is set to Overlay, the network_plugin field can only be set to azure."
     }
+    precondition {
+      condition     = var.ebpf_data_plane != "cilium" || var.network_plugin == "azure"
+      error_message = "When ebpf_data_plane is set to cilium, the network_plugin field can only be set to azure."
+    }
+    precondition {
+      condition     = (var.ebpf_data_plane != "cilium") || (var.network_plugin_mode == "Overlay" && var.pod_subnet_id == null || var.pod_subnet_id != null && var.network_plugin_mode == null)
```

Comment:
You can have `var.network_plugin_mode == "Overlay"` and specify a value for example `var.pod_subnet_id == 10.100.0.0/16`. This is a valid configuration with Cilium but the current precondition would not allow it.

Can you please double check ?

Would it work this one? :
```
(var.ebpf_data_plane != "cilium") || var.network_plugin_mode == "Overlay" || var.pod_subnet_id != null
```
Also here you are just checking `var.pod_subnet_id` for the default node pool, but the additional nodepools are not part for the precondition

Reply:
Yes, I updated the precondition.